ionic-template/views/SubSectionShow.js

```diff
@@ -1,46 +1,35 @@
 import React /*, { useEffect } */ from 'react'
 // import PropTypes from 'prop-types'
-import { compose } from 'redux'
-import { connect } from 'react-redux'
+// import * as Types from 'types'
+// import { useDispatch, useSelector } from 'react-redux'
 import { useParams } from 'react-router-dom'
 import { IonItem, IonLabel, IonList, IonListHeader } from '@ionic/react'
 // import { LoadingModal } from 'components'
-// import { selectors } from '../reducer'
-// import * as actions from '../actions'
+// import { clear%SubSection%, selectors } from '../reducer'
 // import * as apiActions from 'api-actions'
-// import * as Types from 'types'
 
-const propTypes = {
-  // %subSection%: Types.%SubSection%,
-  // fetch%SubSection%: PropTypes.func.isRequired,
-  // clear%SubSection%: PropTypes.func.isRequired,
-}
+const propTypes = {}
 const defaultProps = {}
 
-function %SubSection%Show (
-  // {
-  //   %subSection%,
-  //   fetch%SubSection%,
-  //   clear%SubSection%,
-  // }
-) {
-  const { id } = useParams()
-  //
+function %SubSection%Show () {
+  // const { id } = useParams()
+  // const %subSection% = useSelector(selectors.%subSection%)
+  // const dispatch = useDispatch()
   // useEffect(() => {
-  //  fetch%SubSection%(id)
+  //  dispatch(apiActions.fetch%SubSection%(id))
```

Comment:
is the `apiActions.fetch&SubSection` defined as an async thunk?

So far the only concept that I have not seen applied is `async thunks` and I'm curious if we should incorporate this concept.

Reply:
No, it's not applied as an async thunk. `createAsyncThunk` gives utilities for promise fulfilled / rejected that are _already built_ into `lp-redux-api`. I did some testing with this and we ended up getting a lot of noise in the reducer with extraneous events.

This is the easiest migration path for us to get on to redux toolkit. Eventually, we'll likely either move to RTK Query or to react-query, which will deprecate most, if not all, of `lp-redux-api`. However, that's a much more drastic change and one that I don't think we have the appetite for (yet).

Reply:
I've seen (and written) code where we invoke an API action and, inline, await the result. That API action is already wrapped in `dispatch` by `mapDispatchToProps`. How would that work here? Something like:
```
const result = await dispatch(apiActions.fetchSubsection)
doStuffWithResult()
```

Reply:
@mwislek Yes, that's actually a great callout. You _can_ still do that with this implementation (although it technically would be `const result = await dispatch(apiActions.fetchSubsection())`). If we used `createAsyncThunk`, we'd have to do `const result = await dispatch(apiActions.fetchSubsection()).unwrap()`, which is...confusing to say the least